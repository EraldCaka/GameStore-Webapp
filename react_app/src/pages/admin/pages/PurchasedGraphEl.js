import { Chart, CategoryScale, Title, Tooltip, Legend } from "chart.js/auto";
import { Bar } from "react-chartjs-2";

Chart.register(CategoryScale);
function PurchasedGraphEl({ purchased, games }) {
  console.log(purchased);
  console.log(games[0]);
  const data = {
    labels: [games[0], games[1], games[2]],
    datasets: [
      {
        label: `${purchased[0] + purchased[1] + purchased[2]}`,
        data: [purchased[0], purchased[1], purchased[2]],
        backgroundColor: ["rgba(255,99,132,0.2)"],
        borderColor: ["rgba(255,99,132,1)"],
        borderWidth: 1,
      },
    ],
  };
  const options = {
    maintainAspectRatio: false,
    responsive: true,
    scales: {
      x: {
        grid: {
          display: false,
        },
      },
      y: {
        grid: {
          display: true,
        },
      },
    },
    plugins: {
      legend: {
        display: true,
        position: "top",
      },
    },
  };

  return (
    <div
      style={{
        width: "1750px",
        height: "500px",
        marginLeft: "50px",
        marginTop: "50px",
        marginBottom: "50px",
      }}
    >
      <Bar data={data} options={options} />
    </div>
  );
}

export default PurchasedGraphEl;
