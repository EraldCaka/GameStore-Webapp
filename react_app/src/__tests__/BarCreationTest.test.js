import React from "react";
import { render } from "@testing-library/react";
import PurchasedGraphEl from "../pages/admin/pages/PurchasedGraphEl";
import "jest-canvas-mock";

test("renders PurchasedGraphEl component", () => {
  const purchased = [10, 20, 30];
  const games = ["Game 1", "Game 2", "Game 3"];

  render(<PurchasedGraphEl purchased={purchased} games={games} />);
});
