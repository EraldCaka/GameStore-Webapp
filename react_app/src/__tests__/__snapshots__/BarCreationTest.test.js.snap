// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders PurchasedGraphEl component 1`] = `
<div>
  <div
    style="width: 1750px; height: 500px; margin-left: 50px; margin-top: 50px; margin-bottom: 50px;"
  >
    <canvas
      height="150"
      role="img"
      width="300"
    />
  </div>
</div>
`;
